File: mml-webapp/scripts_upload/mml.R
packages<- function(name){
  if(eval(parse(text=paste("!require(",name,")")))){
    install.packages(name)
    eval(parse(text=paste("library(",name,")")))
  }
}
packages("yardstick")
packages("rminer")
packages('rpart') 
packages("randomForest") 
packages("nnet") 
packages('xgboost') 
packages('e1071') 

#Import the dataset 
data = read.csv('/home/id1019/Documents/iris.csv', sep=',')
columns_names = colnames(data)

#Spliting dataset between features (X) and label (y)
#y-> the last column
Y_name = rev(columns_names)[1]
X_names = rev(columns_names)[-1]
formula <- as.formula(paste(Y_name, "~", paste(X_names, collapse= "+"))) 



# Spliting dataset into training set and test set
train_size = 0.7
indice_train= sample(c(1:nrow(data)), train_size*nrow(data))
train=data[indice_train, ]
test=data[-indice_train, ]
Y_test = test[Y_name] 
X_test = test[X_names]
Y_train = train[Y_name] 
X_train = train[X_names]

results_colnames = c('Algorithm', 'Parameters', 'FrameWork', 'balanced_accuracy', 'recall', 'precision', 'accuracy', 'f1', 'macro_recall', 'macro_precision', 'macro_accuracy', 'macro_f1')
results = matrix(ncol=length(results_colnames),nrow=1)

try({
	algo = rpart(formula, data=train ,method="class", maxdepth = 20)
	Y_pred = predict(algo, X_test,  type="class")
	results = rbind(results, c('DecisionTree', 'max_depth = 20', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, recall_vec(Y_test[,1], Y_pred, estimator="micro") 
	, precision_vec(Y_test[,1], Y_pred, estimator="micro") 
	, accuracy_vec(Y_test[,1], Y_pred) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='ACC') /100 #ACC= equal to micro averaged F1 score (help) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroTPR') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroPRECISION') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroACC') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroF1') /100 
	))
	
	
})

try({
	algo = randomForest(formula, data=train)
	Y_pred = predict(algo, X_test,  type="class")
	results = rbind(results, c('RandomForest', '', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, recall_vec(Y_test[,1], Y_pred, estimator="micro") 
	, precision_vec(Y_test[,1], Y_pred, estimator="micro") 
	, accuracy_vec(Y_test[,1], Y_pred) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='ACC') /100 #ACC= equal to micro averaged F1 score (help) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroTPR') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroPRECISION') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroACC') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroF1') /100 
	))
	
	
})

try({
	algo = multinom(formula, data=train)
	Y_pred = predict(algo, X_test,  type="class")
	results = rbind(results, c('LogisticRegression', '', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, recall_vec(Y_test[,1], Y_pred, estimator="micro") 
	, precision_vec(Y_test[,1], Y_pred, estimator="micro") 
	, accuracy_vec(Y_test[,1], Y_pred) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='ACC') /100 #ACC= equal to micro averaged F1 score (help) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroTPR') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroPRECISION') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroACC') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroF1') /100 
	))
	
	
})

try({
	X_train_matrix = as.matrix(X_train)
	X_test_matrix = as.matrix(X_test)
	label_train = as.integer(Y_train[,1])-1
	label_test = as.integer(Y_test[,1])-1
	num_class = length(levels(Y_train[,1]))
	algo <- xgboost(data = X_train_matrix, label = label_train, nrounds = 100, num_class=num_class ) #nrounds=100 like in python
	Y_pred = predict(algo, X_test_matrix)
	Y_pred = as.factor(levels(Y_train[,1])[Y_pred+1]) 
	results = rbind(results, c('XGboost', '', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, recall_vec(Y_test[,1], Y_pred, estimator="micro") 
	, precision_vec(Y_test[,1], Y_pred, estimator="micro") 
	, accuracy_vec(Y_test[,1], Y_pred) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='ACC') /100 #ACC= equal to micro averaged F1 score (help) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroTPR') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroPRECISION') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroACC') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroF1') /100 
	))
	
	
})

try({
	algo = svm(formula, data=train, type = "nu-classification",kernel = 'radial', gamma=8, cost=2)
	Y_pred = predict(algo, X_test)
	results = rbind(results, c('SVM - Nu-classification', 'kernel = radial, gamma = 8', 'R', bal_accuracy_vec(Y_test[,1], Y_pred) 
	, recall_vec(Y_test[,1], Y_pred, estimator="micro") 
	, precision_vec(Y_test[,1], Y_pred, estimator="micro") 
	, accuracy_vec(Y_test[,1], Y_pred) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='ACC') /100 #ACC= equal to micro averaged F1 score (help) 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroTPR') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroPRECISION') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroACC') /100 
	, mmetric(y=Y_test[,1], x =Y_pred, metric='macroF1') /100 
	))
	
	
})

results_df = data.frame(results, row.names = NULL)
colnames(results_df) =results_colnames
results_df = results_df[-1,]
write.table(results_df,"scripts_upload/results/results_R.csv", sep = ";",dec='.',quote=FALSE, row.names = FALSE)

File: MML-classification-master/org.xtext.example.mml.tests/scripts_upload/mml.py
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split 
from sklearn.metrics import balanced_accuracy_score
from sklearn.metrics import recall_score
from sklearn.metrics import precision_score
from sklearn.metrics import accuracy_score
from sklearn.metrics import f1_score
from sklearn.metrics import recall_score
from sklearn.metrics import precision_score
from sklearn.metrics import f1_score
from sklearn import tree 
from sklearn.ensemble import RandomForestClassifier 
from sklearn.linear_model import LogisticRegression 
import xgboost as xgb 
import sklearn.svm as svm

#Using pandas to import the dataset 
data = pd.read_csv('/home/id1019/Documents/iris.csv', sep=',')
columns_names = data.columns

#Spliting dataset between features (X) and label (y)
Y_name = columns_names[-1]
Y = data[Y_name] 
X = data.drop(columns=Y_name) 



# Spliting dataset into training set and test set
train_size = 0.7
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, train_size=train_size)
results=[['Algorithm', 'Parameters', 'FrameWork', 'balanced_accuracy', 'recall', 'precision', 'accuracy', 'f1', 'macro_recall', 'macro_precision', 'macro_f1']]


try:
	algo = tree.DecisionTreeClassifier(max_depth = 20)
	algo.fit(X_train, Y_train)
	framework_algo = ['DecisionTree', 'max_depth = 20', 'scikit-learn']
	framework_algo.extend([balanced_accuracy_score(Y_test, algo.predict(X_test)) 
	, recall_score(Y_test, algo.predict(X_test),average= 'micro') 
	, precision_score(Y_test, algo.predict(X_test),average= 'micro') 
	, accuracy_score(Y_test, algo.predict(X_test)) 
	, f1_score(Y_test, algo.predict(X_test),average= 'micro') 
	, recall_score(Y_test, algo.predict(X_test),average='macro') 
	, precision_score(Y_test, algo.predict(X_test),average='macro') 
	, f1_score(Y_test, algo.predict(X_test),average='macro') 
	])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')

try:
	algo = RandomForestClassifier()
	algo.fit(X_train, Y_train)
	framework_algo = ['RandomForest', '', 'scikit-learn']
	framework_algo.extend([balanced_accuracy_score(Y_test, algo.predict(X_test)) 
	, recall_score(Y_test, algo.predict(X_test),average= 'micro') 
	, precision_score(Y_test, algo.predict(X_test),average= 'micro') 
	, accuracy_score(Y_test, algo.predict(X_test)) 
	, f1_score(Y_test, algo.predict(X_test),average= 'micro') 
	, recall_score(Y_test, algo.predict(X_test),average='macro') 
	, precision_score(Y_test, algo.predict(X_test),average='macro') 
	, f1_score(Y_test, algo.predict(X_test),average='macro') 
	])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')

try:
	algo = LogisticRegression()
	algo.fit(X_train, Y_train)
	framework_algo = ['LogisticRegression', '', 'scikit-learn']
	framework_algo.extend([balanced_accuracy_score(Y_test, algo.predict(X_test)) 
	, recall_score(Y_test, algo.predict(X_test),average= 'micro') 
	, precision_score(Y_test, algo.predict(X_test),average= 'micro') 
	, accuracy_score(Y_test, algo.predict(X_test)) 
	, f1_score(Y_test, algo.predict(X_test),average= 'micro') 
	, recall_score(Y_test, algo.predict(X_test),average='macro') 
	, precision_score(Y_test, algo.predict(X_test),average='macro') 
	, f1_score(Y_test, algo.predict(X_test),average='macro') 
	])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')

try:
	algo = xgb.XGBClassifier()
	algo.fit(X_train, Y_train)
	framework_algo = ['XGboost', '', 'scikit-learn']
	framework_algo.extend([balanced_accuracy_score(Y_test, algo.predict(X_test)) 
	, recall_score(Y_test, algo.predict(X_test),average= 'micro') 
	, precision_score(Y_test, algo.predict(X_test),average= 'micro') 
	, accuracy_score(Y_test, algo.predict(X_test)) 
	, f1_score(Y_test, algo.predict(X_test),average= 'micro') 
	, recall_score(Y_test, algo.predict(X_test),average='macro') 
	, precision_score(Y_test, algo.predict(X_test),average='macro') 
	, f1_score(Y_test, algo.predict(X_test),average='macro') 
	])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')

try:
	algo = svm.NuSVC( kernel = 'rbf', gamma =8)
	algo.fit(X_train, Y_train)
	framework_algo = ['SVM - Nu-classification', 'kernel = rbf, gamma = 8', 'scikit-learn']
	framework_algo.extend([balanced_accuracy_score(Y_test, algo.predict(X_test)) 
	, recall_score(Y_test, algo.predict(X_test),average= 'micro') 
	, precision_score(Y_test, algo.predict(X_test),average= 'micro') 
	, accuracy_score(Y_test, algo.predict(X_test)) 
	, f1_score(Y_test, algo.predict(X_test),average= 'micro') 
	, recall_score(Y_test, algo.predict(X_test),average='macro') 
	, precision_score(Y_test, algo.predict(X_test),average='macro') 
	, f1_score(Y_test, algo.predict(X_test),average='macro') 
	])
	results.append(framework_algo)
except: print('At least one algorithm has received an error')


pd.DataFrame(results).to_csv("scripts_upload/results/results_python.csv",  sep=";", header=None, index=None)

